Lib/test/test_generators.py

```diff
@@ -112,6 +113,27 @@ def g3(): return (yield from f())
                 gen.send(2)
             self.assertEqual(cm.exception.value, 2)
 
+    def test_generator_resurrect(self):
+        # Test that a resurrected generator still has a valid gi_code
+        resurrected = []
+
+        # Resurrect a generator in a finalizer
+        exec(textwrap.dedent("""
+            def gen():
```

Comment:
Does this need to be in an exec? Could you just `del g` instead?

Reply:
You could then set a `resurrected = None` using a nonlocal instead

Reply:
Yes, it needs to be an exec. Otherwise the generator's code object is referenced by the enclosing code object, which will keep it alive. So the test would still pass, but it wouldn't effectively test the GC behavior.